File: ProblemSolve/763.cs
/********************
 * Partition Labels *
 ********************/

public class Solution {
    public IList<int> PartitionLabels(string s) {
        int[] last = new int[26];
        
        for(int i=0; i<s.Length; ++i){
            last[(int)(s[i] - 'a')] = i;
        }
        
        int idx = 0, anchor = 0;
        List<int> ans = new List<int>();
        
        for(int i=0; i<s.Length; ++i){
            idx = Math.Max(idx, last[(int)(s[i] - 'a')]);
            
            if(i == idx){
                ans.Add(i - anchor + 1);
                anchor = i+1;
            }
        }
        
        return ans;
    }
}

File: ProblemSolve/763.cpp
/********************
 * Partition Labels *
 ********************/

class Solution {
public:
    vector<int> partitionLabels(string s) {
        int last[26] = {0, };
        
        for(int i=0; i<s.size(); ++i){
            last[s[i] - 'a'] = i;
        }
        
        int lastIdx = 0, anchor = 0;
        vector<int> ans;
        
        for(int i=0; i<s.size(); ++i){
            lastIdx = max(lastIdx, last[s[i] - 'a']);
            
            if(i == lastIdx){
                ans.push_back(i - anchor + 1);
                anchor = i + 1;
            }
        }
        
        return ans;
    }
};